apps/ios/Shared/Views/UserSettings/UserProfilesView.swift

```diff
@@ -41,45 +54,103 @@ struct UserProfilesView: View {
 
     private func userProfilesView() -> some View {
         List {
+            if profileHidden {
+                Button {
+                    withAnimation { profileHidden = false }
+                } label: {
+                    Label("Enter password above to show!", systemImage: "exclamationmark.shield")
+                }
+            }
             Section {
-                ForEach(m.users) { u in
+                let users = filteredUsers()
+                ForEach(users) { u in
                     userView(u.user)
                 }
                 .onDelete { indexSet in
                     if let i = indexSet.first {
-                        showDeleteConfirmation = true
-                        userToDelete = i
+                        if m.users.count > 1 && (m.users[i].user.hidden || visibleUsersCount > 1) {
+                            showDeleteConfirmation = true
+                            userToDelete = users[i]
+                        } else {
+                            alert = .cantDeleteLastUser
+                        }
                     }
                 }
 
-                NavigationLink {
-                    CreateProfile()
-                } label: {
-                    Label("Add profile", systemImage: "plus")
+                if searchTextOrPassword == "" {
+                    NavigationLink {
+                        CreateProfile()
+                    } label: {
+                        Label("Add profile", systemImage: "plus")
+                    }
+                    .frame(height: 44)
+                    .padding(.vertical, 4)
                 }
-                .frame(height: 44)
-                .padding(.vertical, 4)
             } footer: {
-                Text("Your chat profiles are stored locally, only on your device.")
+                Text("Tap to activate profile.")
+                    .font(.body)
+                    .padding(.top, 8)
+
             }
         }
         .toolbar { EditButton() }
         .navigationTitle("Your chat profiles")
+        .searchable(text: $searchTextOrPassword, placement: .navigationBarDrawer(displayMode: .always))
```

Comment:
a couple of strange search behaviours:
1. active hidden profile is always present in search results
2. unhiding profile from search results removes it from them (contrary to hiding)

Reply:
unhiding removes them because they no longer have password. I decided not to bother with that gimmick that keeps them, it's not difficult to find them. Will check 1.